src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/UpdateTribunalDecisionRule31MidEvent.java

```diff
@@ -39,7 +39,18 @@ public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callb
         boolean isDecisionAllowed = isDecisionAllowed(asylumCase);
 
         final List<Value> values = new ArrayList<>();
-        if (isDecisionAllowed) {
+        if (asylumCase.read(UPDATED_APPEAL_DECISION, String.class).isPresent()) {
```

Comment:
We don't need to separate isDecisionAllowed and asylumCase.read(UPDATED_APPEAL_DECISION, String.class).isPresent(). They do the same thing.
Also need unit test to cover this part.